src/electron.ts

```diff
@@ -90,6 +89,16 @@ app.on('activate', () => {
   }
 });
 
+function handleWindowsOnClose() {
+  unregisterShortcuts();
+  mainWindowState.saveState(current);
+  let index = windows.findIndex((w) => w.id === current.id);
+  windows.splice(index, 1);
+  current.close();
+  windows.filter(w => w);
```

Comment:
indexes always go from 0 to length - 1, you never need to "reindex". also, `.filter()` returns an array, I believe you meant `windows = windows.filter(w => w);`, but this would make array empty so not sure what did you mean. even if would need to "reindex" array you woud do it with `windows = windows.map(w => w)`... but again, this is not needed.